README.md

```diff
@@ -56,186 +50,416 @@ npm install @dojo/has
+
+The following example passes `css.root` that will be themeable and `css.rootFixed` that cannot be overridden.
+
+```typescript
+import * as css from './styles/myWidget.m.css';
+import { ThemeableMixin, theme } from '@dojo/widget-core/mixins/Themeable';
+
+@theme(css)
+export default class MyWidget extends ThemeableMixin(WidgetBase) {
+	protected render(): DNode {
+		return v('div', { classes: this.classes(css.root).fixed(css.rootFixed) });
+	}
+}
 ```
 
-Include `.tsx` files in the project:
+#### Writing a theme
+
+Themes are typescript modules that export an object that contains css-modules files keyed by a widgets css file name.
 
+```css
+/* myTheme/styles/myWidget.m.css */
+.root {
+	color: blue;
+}
 ```
- "include": [
- 	"./src/**/*.ts",
- 	"./src/**/*.tsx"
- ]
+
+```typescript
+// myTheme/theme.ts
+import * as myWidget from './styles/myWidget.m.css';
+
+export default {
+	'myWidget': myWidget
+}
 ```
 
-Once the project is configured, `tsx` can be used in a widget's `render` function simply by importing the `tsx` function as `import { tsx } from '@dojo/widget-core/tsx';`
+In the above example, the new `root` class provided by the theme for `myWidget` will replace the `root` class that was provided by the original `myWidget.m.css`. This means the `myWidget` will now have color set to blue, but will no longer receive the styles from the `root` class in the original css.
 
-```tsx
-class MyWidgetWithTsx extends WidgetBase<MyProperties> {
-	protected render(): DNode {
-		const { clear, properties: { completed, count, activeCount, activeFilter } } = this;
+#### Applying a theme
 
-		return (
-			<footer classes={this.classes(css.footer)}>
-				<span classes={this.classes(css.count)}>
-					<strong>{`${activeCount}`}</strong>
-					<span>{`${count}`}</span>
-				</span>
-				<TodoFilter activeFilter={activeFilter} />
-				{ completed ? ( <button onclick={clear} /> ) : ( null ) }
-			</footer>
-		);
-	}
+To apply a theme to a widget, simply require it as a module and pass it to a widget via its properties. It is important to ensure that any child widgets created within your widgets render function are passed the `theme` or they will not be themed.
+
+```typescript
+// app.ts
+import myTheme from './myTheme/theme';
+
+// ...
+render() {
+	return w(TabPanel, { theme: myTheme } });
 }
 ```
 
-**Note:** Unfortunately `tsx` is not directly used within the module so will report as an unused import so would be needed to be ignored by linters.
+#### Passing extra classes
+
+Sometimes you may need to apply positioning or layout styles to a child widget. As it is not possible to pass `classes` directly to `WNodes` they provide an `extraClasses` property that can be used to target themeable classes used within its `render` function. In most cases this should only target the `root` class and apply positioning adjustments. The classes passed via `extraClasses` are outside of the theming mechanism and thus will not be effected by a change in `theme`.
```

Comment:
directly to `WNodes` they provide an `extraClasses` property that can be used to target themeable classes used within its `render` function. -> directly to virtualized widget nodes. `WNodes` thus provide an `extraClasses` property to target themeable classes within its `render` function.